src/electionguard/encrypt.py

```diff
@@ -404,21 +411,20 @@ def encrypt_contest(
         chaum_pedersen_nonce,
         contest_description.number_elected,
         nonce=contest_nonce,
+        extended_data=encrypted_contest_data,
     )
 
-    if encrypted_contest is None or encrypted_contest.proof is None:
-        return None  # log will have happened earlier
-
-    if not should_verify_proofs:
-        return encrypted_contest
+    if should_verify_proofs or not encrypted_contest.proof:
+        if encrypted_contest.is_valid_encryption(
+            contest_description_hash, elgamal_public_key, crypto_extended_base_hash
+        ):
+            return encrypted_contest
+        log_warning(
+            f"mismatching contest proof for contest {encrypted_contest.object_id}"
+        )
+        return None
```

Comment:
These changes are to try avoid the default case being `return None` and minimize the number of `return`s